public/app/percona/backup/components/BackupInventory/RestoreBackupModal/RestoreBackupModal.tsx

```diff
@@ -63,13 +69,13 @@ export const RestoreBackupModal: FC<RestoreBackupModalProps> = ({
     if (selectedTimerange && selectedDay) {
       const { startTimestamp, endTimestamp } = selectedTimerange;
       for (let i = 0; i < 24; i++) {
-        if (isSameDay(selectedDay, startTimestamp)) {
-          if (i < getHours(startTimestamp)) {
+        if (isSameDayFromDate(selectedDay, startTimestamp)) {
+          if (i < getHoursFromDate(startTimestamp)) {
```

Comment:
let's move all these `getHoursFromDate(...)` out of the loops, they are always the same

Reply:
Okay I moved them to root to useMemo()